### PR TITLE
Verify quantization configuration on collection creation

### PR DESCRIPTION
**Pull request by timvisee**
The quantization configuration should be verified as well on collection creation.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: lib/segment/src/types.rs [hunk 1]
```diff
@@ -355,6 +355,7 @@ pub struct ScalarQuantizationConfig {
     pub r#type: ScalarType,
     /// Quantile for quantization. Expected value range in (0, 1.0]. If not set - use the whole range of values
     #[serde(skip_serializing_if = "Option::is_none")]
+    // TODO: validation should exclude 0.0
```
Comment (IvanPleshkov) [inline lib/segment/src/types.rs:358]: I guess we can set min value as 0.5. Usual quantile interval - 0.95-1.0

**Review by IvanPleshkov — APPROVED** [review]

File: lib/segment/src/types.rs (hunk 1 above)
Comment (timvisee) [reply lib/segment/src/types.rs:358]: Will do that now!

File: lib/segment/src/types.rs [hunk 2]
```diff
@@ -355,7 +355,7 @@ pub struct ScalarQuantizationConfig {
     pub r#type: ScalarType,
     /// Quantile for quantization. Expected value range in (0, 1.0]. If not set - use the whole range of values
```
Comment (generall) [inline lib/segment/src/types.rs:356]:
```suggestion
    /// Quantile for quantization. Expected value range in (0.5, 1.0]. If not set - use the whole range of values
```

**Review by generall — CHANGES_REQUESTED** [review]

File: lib/segment/src/types.rs (hunk 2 above)
Comment (timvisee) [reply lib/segment/src/types.rs]: Updated! Changed it to `[0.5, 1.0]` instead because the validation range is inclusive on both sides.

**Review by generall — APPROVED** [review]

**Pull request merged**